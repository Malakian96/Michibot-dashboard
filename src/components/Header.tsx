import Image from "next/image";
import { signIn, signOut, useSession } from "next-auth/react";

const Header = () => {
  const { data: session } = useSession();

  return (
    <header className="w-full p-4 bg-discord-dark-light flex justify-between items-center shadow-md">
      <h1 className="text-xl font-bold text-discord-blurple">Michibot</h1>
      {session ? (
        <div className="flex items-center gap-4">
          {session.user?.image && (
            <Image
              src={session.user.image}
              alt="User avatar"
              width={38}
              height={38}
              className="rounded-full"
            />
          )}
          <button
            className="bg-discord-red text-white px-4 py-2 rounded-md hover:bg-opacity-90"
            onClick={() => signOut()}
          >
            Sign Out
          </button>
        </div>
      ) : (
        <button
          className="bg-discord-green text-white px-4 py-2 rounded-md hover:bg-opacity-90"
          onClick={() => signIn("discord")}
        >
          Sign In
        </button>
      )}
    </header>
  );
};

export default Header;
