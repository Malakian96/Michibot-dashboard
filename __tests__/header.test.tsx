import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { useSession, signIn, signOut } from 'next-auth/react';
import Header from '../src/components/Header';

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(),
  signIn: jest.fn(),
  signOut: jest.fn(),
}));

describe('Header', () => {
  it('renders correctly when not signed in', () => {
    (useSession as jest.Mock).mockReturnValue({
      data: null,
      status: 'unauthenticated',
    });

    const { getByText } = render(<Header />);
    expect(getByText('Michibot')).toBeInTheDocument();
    expect(getByText('Sign In')).toBeInTheDocument();
  });

  it('renders correctly when signed in', () => {
    (useSession as jest.Mock).mockReturnValue({
      data: {
        user: {
          name: 'John Doe',
          email: 'john@example.com',
          image: 'https://example.com/avatar.jpg',
        },
      },
      status: 'authenticated',
    });

    const { getByText, getByAltText } = render(<Header />);
    expect(getByText('Michibot')).toBeInTheDocument();
    expect(getByAltText('User avatar')).toBeInTheDocument();
    expect(getByText('Sign Out')).toBeInTheDocument();
  });

  it('calls signIn when Sign In button is clicked', () => {
    (useSession as jest.Mock).mockReturnValue({
      data: null,
      status: 'unauthenticated',
    });

    const { getByText } = render(<Header />);
    fireEvent.click(getByText('Sign In'));
    expect(signIn).toHaveBeenCalledWith('discord');
  });

  it('calls signOut when Sign Out button is clicked', () => {
    (useSession as jest.Mock).mockReturnValue({
      data: {
        user: {
          name: 'John Doe',
          email: 'john@example.com',
          image: 'https://example.com/avatar.jpg',
        },
      },
      status: 'authenticated',
    });

    const { getByText } = render(<Header />);
    fireEvent.click(getByText('Sign Out'));
    expect(signOut).toHaveBeenCalled();
  });
});

describe('HeaderSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
