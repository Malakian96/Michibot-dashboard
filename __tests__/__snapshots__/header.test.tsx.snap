// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuildSelectorSnapshot matches snapshot 1`] = `
<div>
  <header
    class="w-full p-4 bg-discord-dark-light flex justify-between items-center shadow-md"
  >
    <h1
      class="text-xl font-bold text-discord-blurple"
    >
      Michibot
    </h1>
    <div
      class="flex items-center gap-4"
    >
      <img
        alt="User avatar"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="38"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="38"
      />
      <button
        class="bg-discord-red text-white px-4 py-2 rounded-md hover:bg-opacity-90"
      >
        Sign Out
      </button>
    </div>
  </header>
</div>
`;

exports[`HeaderSnapshot matches snapshot 1`] = `
<div>
  <header
    class="w-full p-4 bg-discord-dark-light flex justify-between items-center shadow-md"
  >
    <h1
      class="text-xl font-bold text-discord-blurple"
    >
      Michibot
    </h1>
    <div
      class="flex items-center gap-4"
    >
      <img
        alt="User avatar"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="38"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Favatar.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="38"
      />
      <button
        class="bg-discord-red text-white px-4 py-2 rounded-md hover:bg-opacity-90"
      >
        Sign Out
      </button>
    </div>
  </header>
</div>
`;
